external-authority-sources.md

```diff
@@ -282,3 +282,16 @@ sample for an external source /api/integration/externalsources/orcid/entryValues
   }
 }
 ```
+
+### Search methods
+#### findByEntityType
```

Comment:
Why is this method needed
I'd expect https://github.com/DSpace/RestContract/blob/main/submissionforms.md to define the applicable external sources, so I don't see a use-case for this endpoint

Reply:
I agree that this is a confusing endpoint. I think it minimally needs a description as to why it's needed, and how an External Source can be related to an Entity Type (maybe via a Collection somehow?)

Reply:
Entities are a new concept for DSpace users but it is around in DSpace-CRIS since years. What we learn is that from an user perspective create a publication or create a person are completely separated things.
This mean that for the user it is much intuitive to decide to import a person and decide which of the applicable providers use than starting from a general "create a thing" feature, where everything is proposed together mixing publication, person, journal, orgunit without a specific order.
In DSpace-CRIS we always offer a contextual menù when you start the creation (or import) of an item where you
- first select the entity type
- second select the compatible options (which provider, on which collection)

Reply:
@abollini does this imply the UI will be changed so users need to select an entity type before they can start submission, or where exactly would this be used
If there are assumed design changes, they should be discussed clearly

Reply:
yes, the UI will be adapted to ask first for the entity type and later select among the collections that can deal with the selected entity type.
This behavior will be added to the create new item button in the mydspace and to the import from external source button. In this latest case only the providers able to deal with the selected entity type will be offered to the user and once a record is found and selected to be imported the select collection modal box will be restricted to matching collection.

![dropdown scrollable menu for new submission](https://user-images.githubusercontent.com/1537527/133631428-83542379-041f-46b7-872d-af8a5b902862.png)

![dropdown scrollable menu for import from external source](https://user-images.githubusercontent.com/1537527/133631434-884da855-7bfe-42f5-a8ae-01f3ae2095fa.png)

Please don't mind about the additional third button between the two discussed here, it is an additional dspace-cris feature that is NOT part of what we are planning here to port to dspace

Reply:
@abollini what would happen for repositories without entity types?
* If the repository doesn't use entities at all, I assume the current functionality will be used. But how would Angular be able to differentiate
* And how would the user be able to submit an item without an entity type, in a repository which only has entity types for some collections (e.g. entity types for journals, but all other collections don't use entity types)

Reply:
This is the area that is currently not yet implemented in DSpace-CRIS. The agreed approach https://github.com/DSpace/RestContract/pull/169#issuecomment-911979238 is to create a special entity type "unset" that will be assigned to all the items/collections that have no yet an entity type. The dropdown in the above screen will be improved so that if only a single option is available the dropdown is not shown at all and the only available option is chosen automatically. This will be always the case for DSpace repositories that have not entity types other than the special unset one. Of course we will need to deal with the special "unset" entity type in the routing so that it will be not used to create a custom url for the items and the current behavior will be retained.

Reply:
Thanks for the update @abollini 

So if I understand it correctly:
* If the https://github.com/DSpace/RestContract/blob/main/entitytypes.md is empty, the UI is not changed, there's no extra type selection, and the system will simply return all collections
* If https://github.com/DSpace/RestContract/blob/main/entitytypes.md has types, the dropdown will display all these types, and the "untyped" option
* If the user selects a type, only collections and external sources matching that type will be displayed
* If the user selects untyped, only collection which don't have a type will be displayed (will it also be configurable to display an external source to support untyped?)

I think this can work, but I would recommend to use `null` for the entity type (as it is now), and not set the entity type at all. I don't see any added value in reserved labels there.

Reply:
I'm now looking closely to this discussion (and I know that we have been discussing this weekly, but I didn't had the context). In my opinion, from the point-of-view of a simple user, in terms of the UI perspective, doesn't make much sense giving more complexity to UI.

Probably other approaches makes more sense. Where you have proper generic pages that present you the choice to import content. Like our national scientific profile service, where you are asked to use the available providers. Every provider can import multiple "entity types" at once from the available wizard. Every provider has it's own description (we could use angular translations for that) and it will be clear to the user what is expected to import.

![image](https://user-images.githubusercontent.com/877488/135604921-363d8921-05c6-41f0-87dc-fcc791b0ea73.png)

You will have kind the same in ORCID, for instance, in the Works section, you will have available a list of providers for works:
![image](https://user-images.githubusercontent.com/877488/135605698-61222530-6c65-4a13-a039-c79439bfed8c.png)

the entity type is assumed by the user because the user has the context. So, to me you could have a generic import page, that eventually can present you filtered providers (by supported EntityType). So, to me, to support this, these Endpoints makes much sense. But eventually addressing it the front-end in a different way.